sql/sql_table.cc

```diff
@@ -10130,6 +10130,33 @@ bool mysql_alter_table(THD *thd, const LEX_CSTRING *new_db,
     DBUG_RETURN(true);
   }
 
+  /* MDEV-23485: Change table to merge engine may cause table data lost.
```

Comment:
Is there any earlier hooks into the ha_myisammrg handler that are called so this addition can be in the handler rather than the main alter table implementation? I ask because I assume you've looked. I'll check back on this alter.

Reply:
I have checked there have no hooks are called in the ha_myisammrg handler, except the inplace support check.